### PR TITLE
fix: wrong max_table_id log in remote catalog manager

### LINKED ISSUE #1513: Wrong max_table_id log in remote catalog manager
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Meta, Other

### What happened?

The log in the remote `CatalogManager` is always `Max table id allocated: 1023`:

https://github.com/GreptimeTeam/greptimedb/blob/c8301feed7310937900d03d6f9bd7649f625cc9e/src/catalog/src/remote/manager.rs#L418

Because the `max_file_id` doesn't change at all in `initiate_catalogs` function:

```rust
    /// Fetch catalogs/schemas/tables from remote catalog manager along with max table id allocated.
    async fn initiate_catalogs(&self) -> Result<(HashMap<String, CatalogProviderRef>, TableId)> {
        let mut res = HashMap::new();
        let max_table_id = MIN_USER_TABLE_ID - 1;

        let mut catalogs = self.iter_remote_catalogs().await;
        while let Some(r) = catalogs.next().await {
            let CatalogKey { catalog_name, .. } = r?;
            info!("Fetch catalog from metasrv: {}", catalog_name);
            let catalog = res
                .entry(catalog_name.clone())
                .or_insert_with(|| self.new_catalog_provider(&catalog_name))
                .clone();

            self.initiate_schemas(catalog_name, catalog, max_table_id)
                .await?;
        }

        Ok((res, max_table_id))
    }

```

I think it might be that we want to pass a mutable reference of `max_table_id` into `initiate_schemas`, and change its value in the function. But it looks like we don't implement it correctly.

And I think we should print the log for every catalog in `initiate_catalogs` instead of the current place.It doesn't make sense to print such log for the catalog manager.

### What operating system did you use?

All os sytem

### Relevant log output and stack trace

_No response_

### How can we reproduce the bug?

Described as above.

### PR DESCRIPTION
**Pull request by haohuaijin**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

fix max_table_id bug and update link in CONTRIBUTING.md

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
close #1513

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]
LGTM

Good job, thank u.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -107,6 +107,6 @@ The core team will be thrilled if you participate in any way you like. When you
 
 Also, see some extra GreptimeDB content:
 
-- [GreptimeDB Docs](https://greptime.com/docs)
-- [Learn GreptimeDB](https://greptime.com/products/db)
+- [GreptimeDB Docs](https://docs.greptime.com/)
+- [Learn GreptimeDB](https://greptime.com/product/db)
```
Comment (waynexia) [inline CONTRIBUTING.md:111]: Good eye!

File: src/catalog/src/remote/manager.rs [hunk 2]
```diff
@@ -173,33 +173,38 @@ impl RemoteCatalogManager {
...
 
     /// Fetch catalogs/schemas/tables from remote catalog manager along with max table id allocated.
-    async fn initiate_catalogs(&self) -> Result<(HashMap<String, CatalogProviderRef>, TableId)> {
+    async fn initiate_catalogs(&self) -> Result<HashMap<String, CatalogProviderRef>> {
         let mut res = HashMap::new();
-        let max_table_id = MIN_USER_TABLE_ID - 1;
 
         let mut catalogs = self.iter_remote_catalogs().await;
         while let Some(r) = catalogs.next().await {
+            let mut max_table_id = MIN_USER_TABLE_ID - 1;
```
Comment (waynexia) [inline src/catalog/src/remote/manager.rs:181]: nit-picking: we can define another constant like `MAX_SYS_TABLE_ID` in `src/common/catalog/src/consts.rs`

**Review by waynexia — APPROVED** [review]
Looks pretty to me, thanks @haohuaijin

File: src/catalog/src/remote/manager.rs (hunk 2 above)
Comment (haohuaijin) [reply src/catalog/src/remote/manager.rs]: good idea.

**Pull request merged**